src/spineml/simulator/simulator.cc

```diff
@@ -0,0 +1,820 @@
+
+                // Get post synapse
+                auto postSynapse = synapse.child("LL:PostSynapse");
+                if(!postSynapse) {
+                    throw std::runtime_error("'Synapse' node has no 'PostSynapse' node");
+                }
+
+                // Build synapse population name from name of weight update
+                // **NOTE** this is an arbitrary choice but these are guaranteed unique
+                std::string geNNSynPopName = SpineMLUtils::getSafeName(weightUpdate.attribute("name").value());
+
+                // Find row lengths, indices and max row length associated with sparse connection
+                unsigned int **rowLength = (unsigned int**)getLibrarySymbol(("rowLength" + geNNSynPopName).c_str(), true);
+                unsigned int **ind = (unsigned int**)getLibrarySymbol(("ind" + geNNSynPopName).c_str(), true);
+                const unsigned int *maxRowLength = (const unsigned int*)getLibrarySymbol(("maxRowLength" + geNNSynPopName).c_str(), true);
+
+                // Create connector
+                std::vector<unsigned int> remapIndices;
+                const unsigned int synapseVarSize = Connectors::create(synapse, popSize, trgPopSize,
+                                                                       rowLength, ind, maxRowLength,
+                                                                       basePath, remapIndices);
+
+                // Add postsynapse properties to dictionary
+                addPropertiesAndSizes(basePath, postSynapse, model, geNNSynPopName, trgPopSize, componentSizes);
+                addEventPorts(basePath, postSynapse, componentURLs, componentEventPorts);
+
+                // Add weight update properties to dictionary
+                addPropertiesAndSizes(basePath, weightUpdate, model, geNNSynPopName, synapseVarSize, componentSizes,
+                                      remapIndices.empty() ? nullptr : &remapIndices);
+                addEventPorts(basePath, weightUpdate, componentURLs, componentEventPorts);
+            }
+        }
+    }
+
+    // Call library function to perform final initialize
+    {
+        Timer t("Initialize sparse:");
+        initializeSparse();
+    }
+
+    auto simulation = experiment.child("Simulation");
+    if(!simulation) {
+        throw std::runtime_error("No 'Simulation' node found in experiment");
+    }
+
+    auto eulerIntegration = simulation.child("EulerIntegration");
+    if(!eulerIntegration) {
+        throw std::runtime_error("GeNN only currently supports Euler integration scheme");
+    }
+
+    // Read integration timestep
+    m_DT = eulerIntegration.attribute("dt").as_double(0.1);
+    LOGI << "DT = " << m_DT << "ms";
+
+    // Read duration from simulation and convert to timesteps
+    m_DurationMs = simulation.attribute("duration").as_double() * 1000.0;
+
+
+    // Create directory for logs (if required)
+    const auto logPath = outputPath / "log";
+    filesystem::create_directory(logPath);
```

Comment:
I believe this function returns false if it fails. You might want to test for this.

Reply:
it doesn't distinguish between the case where the output directory exists (normal) and where it can't create it due to lack of disk space or something though so it's not particularly meaningful

Reply:
You could do a check to see if the directory exists beforehand, but it's up to you. If the folder isn't creating I assume you'd just get an error message further down the line anyway.